plugin/src/tryGetObjects.lua

```diff
@@ -0,0 +1,88 @@
+local Fmt = require(script.Parent.Parent.Fmt)
+local Log = require(script.Parent.Parent.Log)
+local PatchSet = require(script.Parent.PatchSet)
+local Promise = require(script.Parent.Parent.Promise)
+
+local function tryGetObjects(instanceMap, apiContext, patch)
+	assert(PatchSet.validate(patch))
+
+	local unappliedPatch = PatchSet.newEmpty()
+
+	-- GetObjects won't help with anything that failed to remove
+	unappliedPatch.removed = patch.removed
+	-- TODO: Implement this? Do we have to?
+	unappliedPatch.added = patch.added
+
+	local assetsToRequest = {}
+	local receiveCallbacks = {}
+
+	Log.trace("tryGetObjects({:#?})", patch)
+
+	-- TODO: added?
+
+	-- GetObjects only create instances, we can't update the properties of existing ones.
+	-- Instead, just create them again, move their children, and replace the instance.
+	for _, update in ipairs(patch.updated) do
+		table.insert(assetsToRequest, update.id)
+		table.insert(unappliedPatch.updated, update)
+
+		receiveCallbacks[update.id] = function(newInstance)
+			table.remove(unappliedPatch.updated, table.find(unappliedPatch.updated, update))
+
+			local oldInstance = instanceMap.fromIds[update.id]
+
+			-- TODO: What if oldInstance is nil?
+			-- TODO: This will dupe children if the rbxm is not empty.
+			for _, oldChild in ipairs(oldInstance:GetChildren()) do
+				oldChild.Parent = newInstance
+			end
+
+			local oldParent = oldInstance.Parent
+			instanceMap:destroyInstance(oldInstance)
+			newInstance.Parent = oldParent
+		end
+	end
+
+	if #assetsToRequest == 0 then
+		return Promise.resolve(unappliedPatch)
+	end
+
+	return apiContext:createAssets(assetsToRequest):andThen(function(assetId)
+		--[[
+			The assets Rojo creates that we will be loading is in the following structure:
+
+			Assuming we requested the following IDs:
+			- DOGE: A doge mesh named Doge
+			- ROCK: A rock mesh named MyPet
+
+			Root: Folder
+			* DOGE: Folder
+				* Doge: MeshPart
+			* ROCK: Folder
+				* MyPet: MeshPart
+		]]
```

Comment:
Instead of creating objects inside a folder, create a model with multiple root objects. The items don't need special names -- they can just be in the same order the plugin requested them in!

Reply:
What would the roots in this example be? `DOGE` and `ROCK` or `Doge` and `MyPet`?

With the latter, is it possible for a .rbxm/.rbxmx that's being imported by Rojo to be multi-rooted?